client/components/data/query-products-list/index.jsx

```diff
@@ -16,13 +16,19 @@ const request =
 		dispatch( requestProductsList( props ) );
 	};
 
-export function useQueryProductsList( { type = 'all', currency, persist } = {} ) {
+export function useQueryProductsList( {
+	type = 'all',
+	currency,
+	persist,
+	product_slug_list,
+} = {} ) {
 	const dispatch = useDispatch();
 
 	// Only runs on mount.
 	useEffect( () => {
-		dispatch( request( { type, currency, persist } ) );
-	}, [ dispatch, type, persist, currency ] );
+		const product_slugs = product_slug_list?.join( ',' );
```

Comment:
I'm wondering if this conversion from array -> string can be done in `requestProductsList` instead. It will keep this layer clean since all functions would accept an array of product slugs, and it would be easier to add type definitions for these in the future. It is difficult to add enforce types on a comma-separated string. :)

Reply:
Good point here, I'll make the necessary updates to have the conversion done in  `requestProductsList`